libraries/ESP32/examples/Zigbee/Zigbee_Light_Switch/Zigbee_Light_Switch.ino

```diff
@@ -0,0 +1,317 @@
+            esp_zb_get_extended_pan_id(extended_pan_id);
+            log_i("Formed network successfully (Extended PAN ID: %02x:%02x:%02x:%02x:%02x:%02x:%02x:%02x, PAN ID: 0x%04hx, Channel:%d, Short Address: 0x%04hx)",
+                     extended_pan_id[7], extended_pan_id[6], extended_pan_id[5], extended_pan_id[4],
+                     extended_pan_id[3], extended_pan_id[2], extended_pan_id[1], extended_pan_id[0],
+                     esp_zb_get_pan_id(), esp_zb_get_current_channel(), esp_zb_get_short_address());
+            esp_zb_bdb_start_top_level_commissioning(ESP_ZB_BDB_MODE_NETWORK_STEERING);
+        } else {
+            log_i("Restart network formation (status: %s)", esp_err_to_name(err_status));
+            esp_zb_scheduler_alarm((esp_zb_callback_t)bdb_start_top_level_commissioning_cb, ESP_ZB_BDB_MODE_NETWORK_FORMATION, 1000);
+        }
+        break;
+    case ESP_ZB_BDB_SIGNAL_STEERING:
+        if (err_status == ESP_OK) {
+            log_i("Network steering started");
+        }
+        break;
+    case ESP_ZB_ZDO_SIGNAL_DEVICE_ANNCE:
+        dev_annce_params = (esp_zb_zdo_signal_device_annce_params_t *)esp_zb_app_signal_get_params(p_sg_p);
+        log_i("New device commissioned or rejoined (short: 0x%04hx)", dev_annce_params->device_short_addr);
+        esp_zb_zdo_match_desc_req_param_t  cmd_req;
+        cmd_req.dst_nwk_addr = dev_annce_params->device_short_addr;
+        cmd_req.addr_of_interest = dev_annce_params->device_short_addr;
+        esp_zb_zdo_find_on_off_light(&cmd_req, user_find_cb, NULL);
+        break;
+    default:
+        log_i("ZDO signal: %s (0x%x), status: %s", esp_zb_zdo_signal_to_string(sig_type), sig_type,
+                 esp_err_to_name(err_status));
+        break;
+    }
+}
+
+static void esp_zb_task(void *pvParameters)
+{
+    esp_zb_cfg_t zb_nwk_cfg = ESP_ZB_ZC_CONFIG();
+    esp_zb_init(&zb_nwk_cfg);
+    esp_zb_on_off_switch_cfg_t switch_cfg = ESP_ZB_DEFAULT_ON_OFF_SWITCH_CONFIG();
+    esp_zb_ep_list_t *esp_zb_on_off_switch_ep = esp_zb_on_off_switch_ep_create(HA_ONOFF_SWITCH_ENDPOINT, &switch_cfg);
+    esp_zb_device_register(esp_zb_on_off_switch_ep);
+    esp_zb_set_primary_network_channel_set(ESP_ZB_PRIMARY_CHANNEL_MASK);
+    ESP_ERROR_CHECK(esp_zb_start(false));
+    esp_zb_main_loop_iteration();
+}
+
+/********************* GPIO functions **************************/
+static QueueHandle_t gpio_evt_queue = NULL;
+
+static void IRAM_ATTR gpio_isr_handler(void *arg)
+{
+    xQueueSendFromISR(gpio_evt_queue, (switch_func_pair_t *)arg, NULL);
+}
+
+static void switch_gpios_intr_enabled(bool enabled)
+{
+    for (int i = 0; i < PAIR_SIZE(button_func_pair); ++i) {
+        if (enabled) {
+            enableInterrupt((button_func_pair[i]).pin);
+        } else {
+            disableInterrupt((button_func_pair[i]).pin);
+        }
+    }
+}
+
+// Tasks for checking the button event and debounce the switch state
+static void switch_button_detected(void *arg)
+{
```

Comment:
This could be something like `onZigBeeButton...()` and add a user callback... etc.
As said... the example are just IDF code, nothing really Arduinish.